vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/cache/LruCache.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.sqlclient.impl.cache;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.sqlclient.impl.PreparedStatement;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * A LRU replacement strategy cache based on {@link java.util.LinkedHashMap} for prepared statements.
+ */
+class LruCache extends LinkedHashMap<String, AsyncResult<PreparedStatement>> {
```

Comment:
why do we cache the `AsyncResult<PreparedStatement>` and not the `PreparedStatement` instead ?

Reply:
It behaves like it used to be. When one prepare statement request fails and the failure would be also cached.

Reply:
I wonder if we might want to improve this because caching failures such as syntax error, network failure, etc... does not help anything from the perspective of users.

Reply:
ok, I didn't remember this part.

However then it does not map correctly to the max prepared statement cache size because this cache size should account for valid statements I think. So perhaps we want to remove failure caching.

Reply:
I have changed this in https://github.com/eclipse-vertx/vertx-sql-client/pull/597/commits/71d8b670e356418360a48c021993bf41b59e0fd0 now.